evm-node.md

```diff
@@ -0,0 +1,149 @@
+         </k>
+         <schedule> SCHED </schedule>
+         <gasPrice> _ => GPRICE </gasPrice>
+         <origin> _ => ACCTFROM </origin>
+         <callDepth> _ => -1 </callDepth>
+         <coinbase> _ => CB </coinbase>
+         <difficulty> _ => DIFF </difficulty>
+         <number> _ => NUMB </number>
+         <gasLimit> _ => GLIMIT </gasLimit>
+         <timestamp> _ => TS </timestamp>
+         <account>
+           <acctID> ACCTFROM </acctID>
+           <nonce> NONCE </nonce>
+           ...
+         </account>
+         <touchedAccounts> _ => SetItem(CB) </touchedAccounts>
+         <activeAccounts> ACCTS </activeAccounts>
+      requires ACCTFROM in ACCTS
+
+    rule <k> runVM(false, ACCTTO, ACCTFROM, _, ARGS, VALUE, GPRICE, GAVAIL, CB, DIFF, NUMB, GLIMIT, TS, _)
+          => #loadAccount ACCTTO
+          ~> #lookupCode ACCTTO
+          ~> #call ACCTFROM ACCTTO ACCTTO GAVAIL VALUE VALUE #parseByteStackRaw(ARGS) false
+          ~> #endVM
+         ...
+         </k>
+         <schedule> SCHED </schedule>
+         <gasPrice> _ => GPRICE </gasPrice>
+         <origin> _ => ACCTFROM </origin>
+         <callDepth> _ => -1 </callDepth>
+         <coinbase> _ => CB </coinbase>
+         <difficulty> _ => DIFF </difficulty>
+         <number> _ => NUMB </number>
+         <gasLimit> _ => GLIMIT </gasLimit>
+         <timestamp> _ => TS </timestamp>
+         <touchedAccounts> _ => SetItem(CB) </touchedAccounts>
+         <activeAccounts> ACCTS </activeAccounts>
+      requires ACCTFROM in ACCTS
+
+    syntax EthereumCommand ::= "#endVM" | "#endCreate"
+ // --------------------------------------------------
+    rule <k> #exception ~> #endVM => #popCallStack ~> #popWorldState ~> #popSubstate ~> 0 </k>
+         <output> _ => .WordStack </output>
+    rule <k> #revert ~> #endVM => #popCallStack ~> #popWorldState ~> #popSubstate ~> #refund GAVAIL ~> 0 </k>
+         <gas> GAVAIL </gas>       
+
+    rule <k> #end ~> #endVM => #popCallStack ~> #dropWorldState ~> #dropSubstate ~> #refund GAVAIL ~> 1 </k>
+         <gas> GAVAIL </gas>
+
+    rule <k> #endCreate => W ... </k> <wordStack> W : WS </wordStack>
+
+    syntax KItem ::= vmResult(return: String,gas: Int,refund: Int,status: Int,selfdestruct: List,logs: List,AccountsCell, touched: List)
+    syntax KItem ::= extractConfig(GeneratedTopCell) [function]
+ // -----------------------------------------------------------
+    rule extractConfig(<generatedTop>... <output> OUT </output> <gas> GAVAIL </gas> <refund> REFUND </refund> <k> STATUS:Int </k> <selfDestruct> SD </selfDestruct> <log> LOGS </log> <accounts> ACCTS </accounts> <touchedAccounts> TOUCHED </touchedAccounts> ... </generatedTop>) => vmResult(#unparseByteStack(OUT),GAVAIL,REFUND,STATUS,Set2List(SD),LOGS,<accounts> ACCTS </accounts>, Set2List(TOUCHED))
+
+    syntax String ::= contractBytes(WordStack) [function]
+ // -----------------------------------------------------
+    rule contractBytes(WS) => #unparseByteStack(WS)
```

Comment:
`contractBytes == #unparseByteStack`? Is the indirection for some compatibility layer?

Reply:
yes